src/Item/MenuItem.php

```diff
@@ -0,0 +1,234 @@
+<?php declare(strict_types=1);
+
+namespace Torr\MenuBundle\Item;
+
+use Symfony\Contracts\Translation\TranslatableInterface;
+use Torr\Rad\Route\LinkableInterface;
+
+class MenuItem
+{
+	// region Fields
+	/**
+	 * The children of the menu item.
+	 *
+	 * @var self[]
+	 */
+	private array $children = [];
+	//endregion
+
+
+	/**
+	 */
+	public function __construct (
+		/**
+		 * The label to display.
+		 * Will be translated using the translation domain given in the renderer.
+		 */
+		private readonly TranslatableInterface|string|null $label = null,
+		/**
+		 * The parent menu item.
+		 */
+		private ?self $parent = null,
+		/**
+		 * The target of this item.
+		 *
+		 * LinkableInterface	-> route
+		 * string				-> direct URI
+		 * null					-> no link
+		 */
+		private readonly LinkableInterface|string|null $target = null,
+		/**
+		 * Whether the item is the currently selected menu item.
+		 */
+		private bool $current = false,
+		/**
+		 * The extra attributes on the menu item.
+		 */
+		private array $extras = [],
+		/**
+		 * Whether the item is virtual (= should be included in the tree but never rendered).
+		 */
+		private readonly bool $virtual = false,
```

Comment:
Do you have a brief example of what we'd be using this for? Is this for something like categories/groups?